cng/sha.go

```diff
@@ -41,6 +41,20 @@ func shaOneShot(id string, p, sum []byte) error {
 	return bcrypt.Hash(h.handle, nil, p, sum)
 }
 
+func MD4(p []byte) (sum [16]byte) {
+	if err := shaOneShot(bcrypt.MD4_ALGORITHM, p, sum[:]); err != nil {
+		panic("bcrypt: MD4 failed")
+	}
+	return
+}
+
+func MD5(p []byte) (sum [16]byte) {
+	if err := shaOneShot(bcrypt.MD5_ALGORITHM, p, sum[:]); err != nil {
```

Comment:
Unfortunate naming issues coming up because all our hashes have been SHAs so far. 😄 The public API looks fine. Arguably since MD5 was used as a secure hash historically, it isn't totally out of place. Not sure it's worth changing, just thought I'd point it out.

Reply:
Lets do it, it makes things clearer.